main/com.spotify/index.js

```diff
@@ -31,534 +31,166 @@
 // ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED
 // OF THE POSSIBILITY OF SUCH DAMAGE.
 "use strict";
-
-const Tp = require('thingpedia');
-const querystring = require('querystring');
-
-const USER_URL = 'https://api.spotify.com/v1/me';
-//const USER_PLAYER_URL = 'https://api.spotify.com/v1/me/player';
-
-
-// Player URLs
-const AVAILABLE_DEVICES_URL = 'https://api.spotify.com/v1/me/player/devices';
-const CURRENTLY_PLAYING_URL = 'https://api.spotify.com/v1/me/player/currently-playing';
-
-const AUDIO_FEATURES_URL = 'https://api.spotify.com/v1/audio-features/';
-const KEY_MAPPINGS = ['C', 'C#/Db', 'D', 'D#/Eb', 'E', 'F', 'F#/Gb', 'G', 'G#/Ab', 'A', 'A#/Bb', 'B'];
-
-// Stateless PUT commands
-const PAUSE_URL = 'https://api.spotify.com/v1/me/player/pause';
-const PLAY_URL = 'https://api.spotify.com/v1/me/player/play';
-
-// Stateless POST commands
-const NEXT_URL = 'https://api.spotify.com/v1/me/player/next';
-const PREVIOUS_URL = 'https://api.spotify.com/v1/me/player/previous';
-
-// Stateful PUT commands
-const SHUFFLE_URL = 'https://api.spotify.com/v1/me/player/shuffle?';
-const REPEAT_URL = 'https://api.spotify.com/v1/me/player/repeat?';
-const SEEK_URL = 'https://api.spotify.com/v1/me/player/seek?';
-const TRACKS_URL = 'https://api.spotify.com/v1/me/tracks?';
-
-// Search URL
-const SEARCH_URL = 'https://api.spotify.com/v1/search?';
-
-//User Library URLs
-// need to replace get userId
-const USER_PLAYLISTS = 'https://api.spotify.com/v1/users/{username}/playlists?';
-const PER_SET = 50;
-
-//Albums
-const ALBUM_URL = 'https://api.spotify.com/v1/albums/{id}/tracks';
-
-
+const Tp = require("thingpedia");
+const USER_URL = "https://api.spotify.com/v1/me";
+const PLAY_URL = "https://api.spotify.com/v1/me/player/play";
+const SEARCH_URL = "https://api.spotify.com/v1/search?";
+const AVAILABLE_DEVICES_URL = "https://api.spotify.com/v1/me/player/devices";
+const ARTIST_ALBUM_URL = "https://api.spotify.com/v1/artists/{id}/albums?";
+const AUDIO_FEATURES_URL = 'https://api.spotify.com/v1/audio-features/?';
+const ALBUM_URL = 'https://api.spotify.com/v1/albums?';
+const TRACK_URL = 'https://api.spotify.com/v1/tracks?';
+const querystring = require("querystring");
 module.exports = class SpotifyDevice extends Tp.BaseDevice {
     static get runOAuth2() {
         return Tp.Helpers.OAuth2({
```

Comment:
This whole block can be replaced with a mixin declared in manifest.tt
```
import mixin from @org.thingpedia.config.oauth2(
  authorize="https://accounts.spotify.com/authorize"^^tt:url,
  get_access_token="https://accounts.spotify.com/api/token"^^tt:url,
  scope=["streaming", "playlist-read-collaborative", "playlist-modify-private", "playlist-read-private", "playlist-modify-public", "user-read-email", "user-read-private", "user-read-playback-state", "user-read-currently-playing", "user-modify-playback-state", "user-read-recently-played", "user-top-read", "user-follow-read", "user-follow-modify", "user-library-read", "user-library-modify"],

  get_profile="https://api.spotify.com/v1/me"^^tt:url,
  profile=["id", "display_name"]
);
```

If you do this, the profile info will be in `this.state.profile.id` and `this.state.profile.display_name`